x/evm/keeper/statedb.go

```diff
@@ -17,6 +17,13 @@ import (
 
 var _ statedb.Keeper = &Keeper{}
 
+func (k *Keeper) NewStateDB(
```

Comment:
I would get rid of this function because it's only a one-line replacement, we use `statedb.New` in other parts of the codebase, and it keeps EVMKeeper more decoupled from StateDB.

Reply:
The rationale's not as obvious with the next change separated out onto #2095.

For context, this is the full function.  
```go
func (evmKeeper *Keeper) NewStateDB(
	ctx sdk.Context, txConfig statedb.TxConfig,
) *statedb.StateDB {
	stateDB := statedb.New(ctx, evmKeeper, txConfig)
	evmKeeper.Bank.ResetStateDB(stateDB)
	return stateDB
}
```

This constructor's meant to be used any time the EVM keeper generates a StateDB to make sure that its bank keeper reference is always pointing to the current `StateDB` for any EVM transaction.